.github/workflows/build.yml

```diff
@@ -58,4 +60,6 @@ jobs:
         if: startsWith(github.ref, 'refs/tags/')
         with:
           body: 'Minor Release (mostly for the automatic download of the jar file)'
-          files: autograder-cmd/target/autograder-cmd.jar
\ No newline at end of file
+          files: |
+              autograder-cmd/target/autograder-cmd.jar
+              autograder-full/target/autograder-full.jar
```

Comment:
Why do we need two binaries? What is in `autograder-full` that is not in `autograder-cmd`?

Reply:
Nothing. The intent was to strip unneeded dependencies and reduce the potential of clashing implementations at runtime, but I'm not sure if that's worth it

Reply:
I don't think that is worth it:

`autograder-cmd.jar`: 50.9MB
`autograder-full.jar`: 50.5MB

Reply:
Actually, the important reason is the thing that @dfuchss mentioned below: `autograder-full` may not contain `autograder-api`, or there will be clashes during dynamic loading. `autograder-cmd` on the other hand needs all dependencies.

Reply:
> Actually, the important reason is the thing that @dfuchss mentioned below: `autograder-full` may not contain `autograder-api`, or there will be clashes during dynamic loading. `autograder-cmd` on the other hand needs all dependencies.

From what I remember that is not a problem. The `URLClassLoader` will always ask the parent classloader first if it knows the class (in case it does, it will load from the parent), before it looks into the JAR. The `autograder-api` is embedded into the JAR, so it will always be known by the parent class loader

Reply:
That is the reason why I had to write the `DynamicClassLoader` which is a reimplementation of `URLClassLoader` with the above behaviour fixed.

Reply:
Seems like you're right, so I'm removing autograder-full

Reply:
Are you sure about the loading order?
What happens if you modify the API e.g., add some classes & change some. Then the API partly consists of the bundled API and the dependend API. Probably, I'd suggest to make sure that the API is not delivered; just to be sure that it won't generate conflicts.

Reply:
Java guarantees the order:

> The ClassLoader class uses a delegation model to search for classes and resources. Each instance of ClassLoader has an associated parent class loader. When requested to find a class or resource, a ClassLoader instance will delegate the search for the class or resource to its parent class loader before attempting to find the class or resource itself.

That's from ClassLoader's Javadoc

Reply:
And your case would be broken anyway, since autograder-core and the downstream user would have the same name referring to different classes